src/Analyzer/Resolve/IdentifierResolver.cpp

```diff
@@ -613,22 +782,67 @@ IdentifierResolveResult IdentifierResolver::tryResolveIdentifierFromStorage(
 
     if (!result_expression)
     {
-        if (can_be_not_found)
-            return {};
-        std::unordered_set<Identifier> valid_identifiers;
-        TypoCorrection::collectTableExpressionValidIdentifiers(
-            identifier,
-            table_expression_node,
-            table_expression_data,
-            valid_identifiers);
+        /// case-insensitive fallback for columns
+        bool ci_columns_enabled = scope.context->getSettingsRef()[Setting::enable_case_insensitive_columns];
 
-        auto hints = TypoCorrection::collectIdentifierTypoHints(identifier, valid_identifiers);
+        std::set<String> ci_col_name_matches;
+        ColumnNodePtr ci_match_column;
+        if (ci_columns_enabled && !table_expression_data.column_name_to_column_node.empty())
+        {
+            String wanted_full_l = Poco::toLower(identifier_without_column_qualifier.getFullName());
+            for (const auto & [col_name, col_node] : table_expression_data.column_name_to_column_node)
+            {
+                if (Poco::toLower(col_name) == wanted_full_l)
+                {
+                    ci_col_name_matches.insert(col_name);
+                    ci_match_column = col_node;
+                }
+            }
+        }
 
-        throw Exception(ErrorCodes::UNKNOWN_IDENTIFIER, "Identifier '{}' cannot be resolved from {}. In scope {}{}",
-            identifier.getFullName(),
-            table_expression_source,
-            scope.scope_node->formatASTForErrorMessage(),
-            getHintsErrorMessageSuffix(hints));
+        if (ci_col_name_matches.size() == 1)
+        {
+            result_expression = ci_match_column;
+            /// If nested path exists (like a.b), resolve it via compound if needed
+            if (!identifier_without_column_qualifier.isShort())
+            {
+                size_t identifier_bind_size = identifier_column_qualifier_parts + 1;
+                result_expression = tryResolveIdentifierFromCompoundExpression(
+                    identifier, identifier_bind_size, result_expression, table_expression_source, scope, can_be_not_found);
+                if (can_be_not_found && !result_expression)
+                    return {};
+                clone_is_needed = false;
+            }
+        }
+        else if (ci_col_name_matches.size() > 1)
+        {
+            throw Exception(
+                ErrorCodes::AMBIGUOUS_IDENTIFIER,
```

Comment:
There is also the `AMBIGUOUS_COLUMN_NAME` error, but probably `AMBIGUOUS_IDENTIFIER` is better for consistency with database and table errors.